determined-ai/determined
# chore: Restrict display names [DET-7356]

## Description

- Prevent non-admin users from appearing as "admin" with display name
- Prevent users from appearing as "determined"
- Prevent users from choosing a display name to copy another user's username / display name
- Prevent users from using non-Latin display names (i.e. ASCII, numbers, Latin alphabet accent marks) (I'm open to making this more flexible, but this avoids someone using invisible spaces, look-alike characters, or the RLM char to disrupt other content and rules)
- In proto/backend, changed PatchUser.DisplayName (null or string) from null.String type to google.protobuf.StringValue (more similar to Model and Workspaces code)

## Test Plan

- Change display name to empty string
- Change display name to dêtérmïned , attempt to change to determined
- Attempt to change display name to Admin / admin
- Confirm previous e2e_tests for display name work

Edit: Trent recommended https://github.com/minimaxir/big-list-of-naughty-strings/blob/master/blns.txt - most of these escaping / scripting strings are already handled by the db driver and react.  I was most worried about the category with strings like "Ṱ̺̺̕o͞ ̷in̝̗͕vokè" , but these get pruned back to a few Latin alphabet markers.

## Checklist
